lib/xander/query/response.ex

```diff
@@ -13,11 +13,23 @@ defmodule Xander.Query.Response do
     end
   end
 
-  # For get_current_block_height
-  defp parse_cbor([@message_response, [@slot_timeline, response]]) do
-    {:ok, response}
+  # This clause parses the response from the get_current_tip query
+  defp parse_cbor([@message_response, [slot_number, %CBOR.Tag{tag: :bytes, value: response}]]) do
+    block_hash = Base.encode16(response, case: :lower)
+    {:ok, {slot_number, block_hash}}
+  end
+
+  # This clause parses the response from the get_current_block_height query
+  defp parse_cbor([@message_response, [@slot_timeline, block_height]]) do
+    {:ok, block_height}
+  end
+
+  # This clause parses the response from get_epoch_number
+  defp parse_cbor([@message_response, [epoch_number]]) do
```

Comment:
This fixes the return type of the get_epoch_number query from a single element array with the epoch number (`[539]`) to the epoch number (`539`)